### ISSUE TITLE
Documenting impact of normalization and gene selection on the quality of the results

### ISSUE THREAD

**Issue by VPetukhov** [broadinstitute/Tangram#3]
Hi Tommaso,
Thanks for the great tool! I'm trying to apply it for deconvolution of Visium data, and am puzzled with few questions. 
First, what's the best way to normalize the data? In the example you mention that you didn't run log-normalization, but I didn't find actual discussion of that in the paper, while normalization was shown to be the most influential step for scRNA-seq analysis. 
Second, what's the best way of selecting genes? I don't expect that we should just blindly take all 30k of them, right? It would be super-valuable to know, whether the method is robust to the gene set or whether one should try different sets until success. 
Finally, would it be too bad if I provide constant density (parameter `d`) for all spots in the tissue? Our stainings are quite messed up, and I don't think segmentation is possible on them.

**Comment by lewlin** [comment]
Hi Viktor,

thank you for your interest and your questions. We are planning to add more notebooks (and expand existing ones) but, meantime, here is a few comments to help get you going:

**Feature (_ie_ gene) selection**
When mapping on Visium, we usually take ~1k genes, selected as markers stratified across cell types. More specifically, 

- We compute cell types and marker genes on scRNA-seq data by running a classic Louvain/Leiden pipeline (see _eg_ the `scanpy` tutorial on PBMC dataset).
- We take the top markers (say, the top 20 genes) for each cell type so that the total number of genes is approximately >1k (some marker genes will be marker of more than one cell type). We call this set of genes the training set.
- (OPTIONAL) In we are interested in resolving specific biological processes (such as pathways, ligand-receptor pairs and so on), we make sure to include relevant genes.
- We then exclude from the training set genes that are not included in the Visium data (as they need to be shared by both datasets).
- (OPTIONAL) We compute and visualize sparsity of the training set between scRNAseq data and Visium data. If a gene is much sparser in a dataset rather than the other, we remove that gene from the training set.
- (OPTIONAL) In general we might want to run additional checks on the genes to make sure they are "high quality". We have a method for this based on active learning but it is not part of the manuscript yet.
- (OPTIONAL) You may want to take out some of the markers to validate your mapping by predicting gene expressions.

 **Applying log-transform**
Although applying `log(1+x)` is critical in computing marker genes, we do not find it essential for mapping. We may, or may not, apply `log(1+x)`. If so, we make sure to apply to both spatial data and scRNAseq data. It is important that both datasets are processed in the same way. Therefore, do not apply `log(1+x)` to scRNAseq data, if you are not applying to the Visium data as well. In the example notebook in which we map VISp nuclei on Visium, we have not applied `log(1+x)`: library-size correction is the only pre-processing step we used.

**Mapping on Visium data**
If you do not have good prior knowledge of the number of cells in each voxel, then deconvolution using Tangram can be challenging. This is what we usually do:
- Rather than inputting a uniform density, we set $\lamba_d = 0$, in order to turn off the density term (the KL divergence term). If we are using the `Mapping` object, we may or may not turn off the entropy regularizer (usually I see how the entropy looks like w/o regularization). If we are using the `MappingConstrained` object, we also turn off the count term (which sets a soft constraint for the "total" number of cells), and the filter regularizer.
- After mapping, we look at probability maps of cell types and prediction of gene expression of test genes not included in training. If they look OK, we call it a good mapping and start looking at cool things (programs distributed in space, reconstruction of low-quality genes etc). _You don't need to deconvolve for achieving this_.
- If you really want to deconvolve your data, but you do not know about cell number within voxels, you can use `MappingConstrained` with settings above and manually inspect the learn filters. That would give you a prediction for the "learned" number of cells. We have not documented this case though in the manuscript, and we are unsure about how well it works.

Ciao,

**Comment by VPetukhov** [comment]
Thank you so much for this detailed answer! I will continue playing with parameters :)
